src/js/ripple/orderbook.js

```diff
@@ -439,19 +439,22 @@ OrderBook.prototype.setFundedAmount = function(offer, fundedAmount) {
     return offer;
   }
 
-  var isOfferGetsExceeded = Amount.from_json(fundedAmount)
+  var isOfferGetsExceeded = Amount.from_json(
+    this._currencyGets.is_native()
+    ? fundedAmount
+    : fundedAmount + OrderBook.IOU_SUFFIX
```

Comment:
If I understand correctly, shouldn't something like this work:

```
if (this._currencyGets.is_native()) {
    return fundedAmount;
} else {
    return {value: fundedAmount, currency: this._currencyGets.toString(), issuer: this._issuerGets.toString()};
}
```

That doesn't seem like too much trouble to get rid of a really ugly hack. And I point out the duplication because this bug is a result of missing one case. It would be less likely to be missed if there was a pattern of calling a conversion function on all incoming parameters at the top of the function.